src/rbr/RBR.java

```diff
@@ -88,15 +88,15 @@ public void addRoutingOptions(ArrayList<ArrayList<Path>> paths) {
 	}
 
 	// Do output combinations
-	private static List<Set<Character>> getOutputCombinations() {
-		List<Set<Character>> oPComb = new ArrayList<>();
-		char[] op = "ENSW".toCharArray();
-
-		for (int m = 1; m != 1 << op.length; m++) {
-			Set<Character> a = new HashSet<>();
-			for (int i = 0; i != op.length; i++) {
+	private static List<Set> allCombinationsOf(Set op) {
+		List<Set> oPComb = new ArrayList<>();
+		for (int m = 1; m != 1 << op.size(); m++) {
+			Set<Object> a = new HashSet<>();
+                        Iterator<Object> iter = op.iterator();

```

Comment:
I would use a toArray() outside the forS instead of an Iterator. The sequence of the iterator is unpredictable and we don't know if it can change between calls. If that happens then this method will fail.